### PR TITLE
fmt: Fix shared, style consistency, cache cmake and proper install bin folder

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **fmt/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

closes https://github.com/conan-io/conan-center-index/issues/1287
closes https://github.com/conan-io/conan-center-index/issues/1288

Also cache CMake and few minor things (style consistency, CMake Verbose disabled, no cross_building in test_package)

### CODE REVIEW

File: recipes/fmt/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,5 @@
+cmake_minimum_required(VERSION 3.1.2)
```
Comment (prince-chrismc) [inline recipes/fmt/all/test_package/CMakeLists.txt:1]: could still be 2.8.x

Comment (SpaceIm) [reply recipes/fmt/all/test_package/CMakeLists.txt:1]: According to conan documentation, `conan_basic_setup requires(TARGETS)` requires 3.1.2: https://docs.conan.io/en/latest/reference/generators/cmake.html#macros-available-in-conanbuildinfo-cmake

Comment (prince-chrismc) [reply recipes/fmt/all/test_package/CMakeLists.txt:1]: Ahhh very cool, I'll have to read what that does =p

**Comment by SpaceIm** [comment]
In package:
```python
        if self.options.with_fmt_alias:
            self.cpp_info.defines.append("FMT_STRING_ALIAS=1")
        if self.options.header_only:
            self.cpp_info.defines = ["FMT_HEADER_ONLY"]
```
If header_ony and with_fmt_alias, "FMT_STRING_ALIAS=1" is removed. Sounds like an issue, isn't it?

**Comment by uilianries** [comment]
@SpaceIm Indeed it's a bug. There is no relation between both defines.

The FMT_STRING_ALIAS is defined by the user only, the definition FMT_HEADER_ONLY is mandatory, but won't configure FMT_STRING_ALIAS.

https://github.com/fmtlib/fmt/blob/d1d653d8955e1948e904e106e196a3d225b579bc/include/fmt/format.h#L3637

**Comment by SpaceIm** [comment]
Ok. One more question: does FMT_STRING_ALIAS changes binaries and includes files? If not, this option should not contribute to package id.
I'm wondering if we should keep this option. I think it should be left to consumer to define ii (there are others macros like this I think, and they don't have any option in conan recipe).

**Comment by uilianries** [comment]
> does FMT_STRING_ALIAS change binaries and includes files? If not, this option should not contribute to package id.

No, it doesn't. You can remove it from the package id.

https://github.com/fmtlib/fmt/search?q=FMT_STRING_ALIAS&unscoped_q=FMT_STRING_ALIAS

> I'm wondering if we should keep this option. I think it should be left to consumer to define it (there are others macros like this I think, and there are no options for them in conan recipe).

Yes, once available in CCI, you can't remove an option, or you can break some users. Backward compatibility is important.

**Pull request merged**